components/places/src/observation.rs

```diff
@@ -119,6 +125,11 @@ impl VisitObservation {
         self
     }
 
+    pub fn with_preview_image_url(mut self, v: impl Into<Option<String>>) -> Self {
```

Comment:
I think this should explicitly be `Option<Url>` - I haven't checked, but I suspect the code you have would *allow* you pass an `Option<Url>` but would also let you pass an `Option<String>` with an invalid Url, which would be bad. IOW, this should be more like `new` which takes a `Url` rather than in `Into<String>`

Reply:
I couldn't figure out how to pass in garbage inputs with this being Into<Option<Url>>, so kept it as-is... Afaik we only use this builder in tests anyway..?